batch.go

```diff
@@ -0,0 +1,94 @@
+package iavl
+
+import (
+	dbm "github.com/cosmos/cosmos-db"
+)
+
+// BatchWithFlusher is a wraper
+// around batch that flushes batch's data to disk
+// as soon as the configurable limit is reached.
+type BatchWithFlusher struct {
+	db                      dbm.DB    // This is only used to create new batch
+	batch                   dbm.Batch // Batched writing buffer.
+	batchSizeFlushThreshold int       // The maximum size of the batch in bytes before it gets flushed to disk
+}
+
+var _ dbm.Batch = BatchWithFlusher{}
+
+// this value is used to estimate the additional size when adding an entry to a batch
+// additional size = key length + value length + possibleEntryOverHead
+var possibleEntryOverHead = 100
+
+// Ethereum has found that commit of 100KB is optimal
+var defaultFlushThreshold = 100000
```

Comment:
Can we make this a configurable parameter with the default value set to this? We should also link the source of this information.

Let me know if I can help with finding the source

Reply:
https://github.com/ethereum/go-ethereum/pull/15115 I think the refrence is here

Reply:
Amazing, thanks!